sdk/python/feast/feature_store.py

```diff
@@ -114,30 +114,39 @@ class FeatureStore:
     repo_path: Path
     _registry: BaseRegistry
     _provider: Provider
-    _go_server: "EmbeddedOnlineFeatureServer"
+    _go_server: Optional["EmbeddedOnlineFeatureServer"]
 
     @log_exceptions
     def __init__(
         self,
         repo_path: Optional[str] = None,
         config: Optional[RepoConfig] = None,
+        fs_yaml_file: Optional[Path] = None,
     ):
         """
         Creates a FeatureStore object.
 
         Raises:
             ValueError: If both or neither of repo_path and config are specified.
         """
-        if repo_path is not None and config is not None:
-            raise ValueError("You cannot specify both repo_path and config.")
-        if config is not None:
+        if fs_yaml_file is not None and config is not None:
```

Comment:
hmm shouldn't this continue to check `repo_path`? as well as the logic below? since we still need exactly one of `repo_path` and `config`, and if `repo_path` is specified, there's an option for `fs_yaml_file` to be specified as well

Reply:
I think we actually need repo path regardless, and one of config or fs_yaml_path.

... I may change the FS constructor cause I feel like we should be able to only pass in in-memory objects to construct it. Dealing with the file system etc should not be something the FS has to care about (ideally)